src/resource/solidDataset.test.ts

```diff
@@ -1803,3 +1810,377 @@ describe("createContainerInContainer", () => {
+      thing,
+      "https://arbitrary.vocab/predicate",
+      "Arbitrary string"
+    );
+    const undoneDataset = setThing(changedDataset, undoneThing);
+
+    expect(getChangeLogAsMarkdown(undoneDataset)).toBe(
+      "## Changes compared to https://some.pod/resource\n\n" +
+        "This SolidDataset has not been modified since it was fetched from https://some.pod/resource.\n"
+    );
+  });
+
+  it("returns a readable version of local changes to a SolidDataset", () => {
+    let thing1 = mockThingFrom("https://some.pod/resource#thing1");
+    thing1 = addStringNoLocale(
+      thing1,
+      "https://some.vocab/predicate",
+      "Some string"
+    );
+    thing1 = addStringNoLocale(
+      thing1,
+      "https://some.vocab/predicate",
+      "Some other string"
+    );
+    let thing2 = mockThingFrom("https://some.pod/resource#thing2");
+    thing2 = addStringNoLocale(
+      thing2,
+      "https://some.vocab/predicate",
+      "Some string"
+    );
+    thing2 = addStringNoLocale(
+      thing2,
+      "https://some.vocab/predicate",
+      "Some other string"
+    );
+    let datasetWithSavedThings = setThing(
+      mockSolidDatasetFrom("https://some.pod/resource"),
+      thing1
+    );
+    datasetWithSavedThings = setThing(datasetWithSavedThings, thing2);
+
+    // Pretend that datasetWithSavedThing was fetched from the Pod with its current contents:
+    datasetWithSavedThings.internal_changeLog = {
+      additions: [],
+      deletions: [],
+    };
+    let changedThing1 = addStringNoLocale(
+      thing1,
+      "https://some.vocab/predicate",
+      "Yet another string"
+    );
+    changedThing1 = removeStringNoLocale(
+      changedThing1,
+      "https://some.vocab/predicate",
+      "Some other string"
+    );
+    let changedThing2 = removeStringNoLocale(
+      thing2,
+      "https://some.vocab/predicate",
+      "Some string"
+    );
+    changedThing2 = removeStringNoLocale(
+      changedThing2,
+      "https://some.vocab/predicate",
+      "Some other string"
+    );
+    let changedDataset = setThing(datasetWithSavedThings, changedThing1);
+    changedDataset = setThing(changedDataset, changedThing2);
+
+    let newThing = createThing({ name: "thing3" });
+    newThing = addStringNoLocale(
+      newThing,
+      "https://some.vocab/predicate",
+      "Some string"
+    );
+    changedDataset = setThing(changedDataset, newThing);
+
+    expect(getChangeLogAsMarkdown(changedDataset)).toBe(
+      "## Changes compared to https://some.pod/resource\n\n" +
+        "### Thing: https://some.pod/resource#thing1\n\n" +
+        "Property: https://some.vocab/predicate\n" +
+        '- Removed: "Some other string" (string)\n' +
+        '- Added: "Yet another string" (string)\n\n' +
+        "### Thing: https://some.pod/resource#thing2\n\n" +
+        "Property: https://some.vocab/predicate\n" +
+        '- Removed: "Some string" (string)\n' +
+        '- Removed: "Some other string" (string)\n\n' +
```

Comment:
Thought about it shortly but didn't have a strong preference either way - I think I went with the current setup because it might make it easier to compare if a value was replaced by another value. But given that that will usually only happen when there's a single value, I guess that would be about as easy. Should be easy to change, I'll give that a shot.

Reply:
Yep, not hard to do at all. Does 30b50f5c06bb93be44d3254ec8cbee271484f1f7 look like you'd expect?

Reply:
Yep, perfect.